### DISCUSSION TITLE
How do I change 127.0.0.1:port to 0.0.0.0:port?

### DISCUSSION THREAD

**Discussion by xtj2020** [tensorchord/envd#922] [Q&A] [answered]
I want to access remotely service in local area,but envd seem use 127.0.0.1:port to map port.
What shoud I do if I want 0.0.0.0:port to provide service? Just like the second ways in the image.
![79](https://user-images.githubusercontent.com/45588939/190992199-2cf8e086-7d2a-4691-a54e-991f08101893.jpg)

**Comment by kemingy** [comment, marked as answer]
Currently, `envd` doesn't support exposure to 0.0.0.0.

You can access your remote port by using:
1. ssh remote port forwarding
2. using port-expose tools like [localtunnel](https://github.com/localtunnel/localtunnel)

**Reply by gaocegege** [reply]
Here is the doc for this https://zh.envd.tensorchord.ai/guide/ide.html#%E5%B0%86%E4%BD%A0%E7%9A%84%E6%9C%8D%E5%8A%A1%E6%9A%B4%E9%9C%B2%E5%87%BA%E5%8E%BB

**Reply by xtj2020** [reply]
Thank you for provide `envd` such awesome tools! And I get the answer.

I used ZeroTier to combines some computes as local area Net.

It's convenient for me if could exposure  0.0.0.0.

As I could provide local area service without considering security.

**Reply by gaocegege** [reply]
I created an issue to keep track of it #923 

Thanks for raising the discussions!

**Reply by gaocegege** [reply]
BTW, Are you using envd on Linux?

**Reply by xtj2020** [reply]
> BTW, Are you using envd on Linux?

Yes,standalone Linux,not WSL.

**Comment by gaocegege** [comment]
Maybe we could support this in build.envd
```python
config.ssh(host="0.0.0.0")
config.jupyter(host="0.0.0.0")
```
